packages/components/src/tabs/tablist.tsx

```diff
@@ -50,9 +152,28 @@ export const TabList = forwardRef<
 		<Ariakit.TabList
 			ref={ ref }
 			store={ store }
-			render={ <TabListWrapper /> }
+			style={
+				{
+					'--indicator-left': `${ indicatorPosition.left }px`,
+					'--indicator-top': `${ indicatorPosition.top }px`,
+					'--indicator-width': `${ indicatorPosition.width }px`,
```

Comment:
I think that's just a consequence of the min-width of the container being larger than the natural width of the tab list. You can see this example where it also happens:

<img width="564" alt="image" src="https://github.com/WordPress/gutenberg/assets/8862967/1915c625-d7b8-49ab-bcf4-9bdef0a51a1f">

The only difference is that in your example, by pure luck, the width is _almost_ the same as the container, so I see how it might feel like there's some weird padding, but it's simply a natural result of sizing.